openedx/core/djangoapps/safe_sessions/middleware.py

```diff
@@ -768,6 +768,39 @@ def _get_encrypted_request_headers(request):
         return encrypt_for_log(str(request.headers), getattr(settings, 'SAFE_SESSIONS_DEBUG_PUBLIC_KEY', None))
 
 
+class EmailVerificationMiddleware(MiddlewareMixin):
```

Comment:
What do you think of something like EmailChangeMiddleware or EmailChangeCheckSessionMiddleware? Note the current name makes me think of the email activation, but maybe that’s just me.

Reply:
You are correct! We will update it with a meaningful name.